### PR TITLE
Feature/rating db

### PR DESCRIPTION
**Pull request by AlessioVerardo**
Hello. 
This PR is a building block for the work of Mathieu and Stephane. It allows us to communicate easily with the database to add/delete/update/get ratings (or specific rating). 

Since I had time, I also refactor the EventDatabase entirely so that we only use the generic method created by Neygo (the objective is that, at the end of the project, we only use these generic methods so that we can get rid of the listener in the database class (more clean code)). 

The refactoring I made make me change the test, we got a 100% coverage  over the Events related method and I'm waiting for a feature that will develop Robin to achieve 100% rating on the Rating-related event. 

I also corrected many warnings that appeared in the stacktrace on Cirrus for example, making the debugging easier ! 

Do not hesitate to make any comment ! 
Cheers,

### CODE REVIEW

File: app/src/androidTest/java/com/github/sdpteam15/polyevents/view/activity/ItemRequestActivityTest.kt [hunk 1]
```diff
@@ -216,26 +216,3 @@ class ItemRequestActivityTest {
...
-            val windowToken = item.decorView.windowToken
-            val appToken = item.decorView.applicationWindowToken
-
-            if (windowToken == appToken) {
-                return true
-            }
-        }
-        return ++currentFailures >= 5
-    }
-}
```
Comment (Cell00phane) [inline app/src/androidTest/java/com/github/sdpteam15/polyevents/view/activity/ItemRequestActivityTest.kt:241]: Why did you remove this test?

File: app/src/main/java/com/github/sdpteam15/polyevents/model/database/remote/objects/EventDatabase.kt [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+                    if (limit != null)
+                        matcher.match(it).limit(limit)
+                    else
+                        matcher.match(it)
+                } else {
+                    if (limit != null)
+                        it.limit(limit)
+                    else
+                        it
+                }
```
Comment (Cell00phane) [inline app/src/main/java/com/github/sdpteam15/polyevents/model/database/remote/objects/EventDatabase.kt:52]: Now that I see it we could probably replace this by:

```
var query = it
if (matcher != null) query = matcher.match(it)
if (limit != null) query = query.limit(limit)
query
```

File: app/src/main/java/com/github/sdpteam15/polyevents/view/fragments/LoginFragment.kt [hunk 3]
```diff
@@ -61,14 +61,15 @@ class LoginFragment : Fragment() {
...
                         val application = requireActivity().application as PolyEventsApplication
                         (application).applicationScope.launch(Dispatchers.IO) {
                             LocalDatabase.populateDatabaseWithUserEvents(
-                                    application.database.eventDao(),
-                                    application.applicationScope)
+                                application.database.eventDao(),
+                                application.applicationScope)
                         }
+
+                        addIfNotInDB()
```
Comment (Cell00phane) [inline app/src/main/java/com/github/sdpteam15/polyevents/view/fragments/LoginFragment.kt:72]: Great catch! Sorry about that.

**Review by Cell00phane — CHANGES_REQUESTED** [review]
Great refactoring as usual! The warnings should not appear now, just have to watch out if we do bad casts ourselves, we won't have warnings for them anymore. Otherwise good job!

File: app/src/androidTest/java/com/github/sdpteam15/polyevents/view/activity/ItemRequestActivityTest.kt (hunk 1 above)
Comment (AlessioVerardo) [reply app/src/androidTest/java/com/github/sdpteam15/polyevents/view/activity/ItemRequestActivityTest.kt:241]: I removed the tests because (I believe) @cpittet tried to test if the toast were shown or not. I believe that this doesn't work and we never use it (Android Studio tells me to remove it since this class was not used in the code) :)

File: app/src/main/java/com/github/sdpteam15/polyevents/model/database/remote/objects/EventDatabase.kt (hunk 2 above)
Comment (AlessioVerardo) [reply app/src/main/java/com/github/sdpteam15/polyevents/model/database/remote/objects/EventDatabase.kt:52]: I totally agree, it has been changed in the last version !

**Review by Cell00phane — APPROVED** [review]

File: app/src/test/java/com/github/sdpteam15/polyevents/database/adapter/RatingAdapterTest.kt [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    val userID = "userId"
+    val ratingId = "ratingID"
+    val eventId = "eventId"
+    val feedback = "feedback"
+    val rate = 3.5F
+
+    lateinit var rating: Rating
+
+    @Before
+    fun setupProfile() {
```
Comment (Cell00phane) [inline app/src/test/java/com/github/sdpteam15/polyevents/database/adapter/RatingAdapterTest.kt:23]: Oh and don't forget to rename to setupRating

File: app/src/main/java/com/github/sdpteam15/polyevents/model/database/remote/objects/EventDatabase.kt [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+            if (it.value) {
+                val m = rating.fold(
+                    Pair(0.0F, 0),
+                    { a, b ->
+                        Pair(
+                            (a.first * a.second + b.rate!!) / (a.second + 1),
+                            a.second + 1
+                        )
+                    })
+                mean.postValue(m.first, it.sender)
```
Comment (Feez2403) [inline app/src/main/java/com/github/sdpteam15/polyevents/model/database/remote/objects/EventDatabase.kt:97]: dont you need to end.postValue(true) here ?

File: app/src/main/java/com/github/sdpteam15/polyevents/view/fragments/MapsFragment.kt [hunk 6]
```diff
@@ -52,18 +52,18 @@ class MapsFragment : Fragment(), OnMapReadyCallback, OnPolylineClickListener,
     var timerHeatmap: Timer? = null
 
     override fun onCreateView(
-        inflater: LayoutInflater,
-        container: ViewGroup?,
-        savedInstanceState: Bundle?
+            inflater: LayoutInflater,
+            container: ViewGroup?,
+            savedInstanceState: Bundle?
```
Comment (Feez2403) [inline app/src/main/java/com/github/sdpteam15/polyevents/view/fragments/MapsFragment.kt:57]: not sure why there are these double tabulations all over your code, does ctrl+alt+L fix it ?

**Review by Feez2403 — APPROVED** [review]
Nice refactor!

File: app/src/main/java/com/github/sdpteam15/polyevents/view/fragments/MapsFragment.kt (hunk 6 above)
Comment (AlessioVerardo) [reply app/src/main/java/com/github/sdpteam15/polyevents/view/fragments/MapsFragment.kt:57]: Ctrl + alt+L create them :)

File: app/src/main/java/com/github/sdpteam15/polyevents/model/database/remote/objects/EventDatabase.kt (hunk 5 above)
Comment (AlessioVerardo) [reply app/src/main/java/com/github/sdpteam15/polyevents/model/database/remote/objects/EventDatabase.kt:97]: Totally, I will fix them in the next version

File: app/src/main/java/com/github/sdpteam15/polyevents/view/fragments/MapsFragment.kt (hunk 6 above)
Comment (Feez2403) [reply app/src/main/java/com/github/sdpteam15/polyevents/view/fragments/MapsFragment.kt:57]: It's strange, when I took your branch it deleted them by reformating code, maybe you changed some settings in android studio?
